### PR TITLE
Create system tests for Standalone Code Security billing

### PR DESCRIPTION
**Pull request by jandro996**
## Motivation

Create system tests for Standalone Code Security billing

JIRA: [APPSEC-55330]

## Changes

- The `Test_AppSecStandalone_UpstreamPropagation` class, previously used for testing AppSec cases, has been refactored into a new base class: `AsmStandalone_UpstreamPropagation_Base`.

- The test cases are identical for both AppSec and IAST, with the following differences:

  - For AppSec, when an ASM event needs to be generated during the requestdownstream call, the header "User-Agent": "Arachni/v1" is added.
  - For IAST, the vulnerablerequestdownstream endpoint is called, which triggers a vulnerability.

- Two new classes extend from the base class:
  - `Test_AppSecStandalone_UpstreamPropagation` runs appsec_standalone scenario.
  - `Test_IastStandalone_UpstreamPropagation` runs a new iast_standalone scenario.

- Java endpoints were added 

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

[APPSEC-55330]: https://datadoghq.atlassian.net/browse/APPSEC-55330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: utils/_context/_scenarios/__init__.py [hunk 1]
```diff
@@ -335,6 +335,18 @@ def all_endtoend_scenarios(test_object):
         scenario_groups=[ScenarioGroup.APPSEC],
     )
 
+    iast_standalone = EndToEndScenario(
```
Comment (cbeauchesne) [inline utils/_context/_scenarios/__init__.py:338]: Could you add this new scenario  in `.github/workflows/run-end-to-end.yml`?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: utils/_context/_scenarios/__init__.py (hunk 1 above)
Comment (jandro996) [reply utils/_context/_scenarios/__init__.py:338]: Do you mean add it to `scenario_groups.yml` and `run-end-to-end.yml` as it was done for appsec_standalone?
I've done it in https://github.com/DataDog/system-tests/pull/3313/commits/3363cb5d53b6dea02f7f371d6282de56eaab7580

**Review by cbeauchesne — COMMENTED** [review]
New scenario is failing on node : 


image

**Comment by jandro996** [comment]
> New scenario is failing on node :
> 
> image

Thanks for the advice, I made a mistake in my last node yaml file merge

**Review by cbeauchesne — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**